src/phases/execution/ClusteredCollection.yml

```diff
@@ -134,17 +133,46 @@ Actors:
         Operations:
         - OperationName: find
           OperationCommand:
-            Filter: {a: {^FastRandomString: {length: 6, alphabet: "0123456789"}}}
+            Filter: {a: *RandomSecondaryKey}
 
+# Phase 6: Range lookup with both primary and secondary key
+- Name: RangeLookupByPrimaryAndSecondaryIndexes
+  Type: CrudActor
+  Database: *Database
+  Threads: 16
+  Phases:
+    OnlyActiveInPhases:
+      Active: [6]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {
+              _id: {^Choose: {
+                from: [
+                  {$gte: *RandomId},
+                  {$lte: *RandomId}
+                ]
+              }},
+              a: {^Choose: {
```

Comment:
Am I understanding correctly that before your planned change, the query will make use of a single secondary index, and after your planned change, the query will use the index intersection of both indices?

Reply:
No.

Planner will pick either clustered index with filter on 'a' or secondary index, fetch and filter on '_id'.

In general it picks one which encounters matching elements more often during trial stage.

I didn't check how we generate index intersection plans, but I bet it will be much harder than what I already did because clustered index scan is still a collection scan, so I don't if our index intersection even supports that.

Reply:
I can look into it, but I think it should be a separate ticket to enable clustered index + secondary index intersection.